repository/src/main/java/org/alfresco/repo/serviceaccount/ServiceAccountRegistryImpl.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * #%L
+ * Alfresco Data model classes
+ * %%
+ * Copyright (C) 2005 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.serviceaccount;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.alfresco.service.cmr.security.PermissionService;
+import org.alfresco.util.PropertyCheck;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Processes the <b>alfresco-global</b> properties file and applies a naming convention to distinguish the service
+ * account's name and role.
+ * <p>
+ * The naming convention adheres to the following format:
+ * <p>
+ * <pre>
+ *   {@code
+ *     serviceaccount.role.<service-account-name>=<service-account-role>
+ *   }
+ * </pre>
+ * <p>
+ * Please note, any property with an invalid role value will be disregarded and the corresponding service account
+ * will not be registered.
+ * <p>
+ * For instance, to register a service account named 'custom-app-sa' with the 'Editor' role (which allows it to
+ * update node properties), the following should be defined in the <b>alfresco-global</b> properties file:
+ * <ul>
+ *   <li>serviceaccount.role.custom-app-sa=EDITOR_SERVICE_ACCOUNT</li>
+ *   <li>or</li>
+ *   <li>serviceaccount.role.custom-app-sa=ROLE_EDITOR_SERVICE_ACCOUNT</li>
+ * </ul>
+ *
+ * @author Jamal Kaabi-Mofrad
+ */
+public class ServiceAccountRegistryImpl implements ServiceAccountRegistry, InitializingBean
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ServiceAccountRegistryImpl.class);
+
+    public static final String KEY_PREFIX = "serviceaccount.role.";
+
+    private Properties globalProperties;
+    private final ConcurrentMap<String, String> saRoleMap = new ConcurrentHashMap<>();
```

Comment:
This map will be local to a specific Repository instance, however since they're all populated at "init" time from the global properties, all instances will also retain the same information assuming all nodes are configured the same way. Meaning we don't need any changes in ent-repo to handle this registry as a shared cache.

Can you confirm this?

Reply:
That's correct; creating a clustered cache is outside the scope of this task. Customers will need to restart their instances to access up-to-date values if they have modified them via the JMX bean.